src/main/java/seedu/duke/Duke.java

```diff
@@ -4,50 +4,49 @@
 import seedu.command.CommandResult;
 import seedu.equipment.EquipmentManager;
 import seedu.parser.Parser;
+import seedu.ui.TextUi;
 
 import java.util.Scanner;
 
 public class Duke {
     private EquipmentManager equipmentInventory = new EquipmentManager();
+    private TextUi ui;
 
     /**
      * Main entry-point for the java.duke.Duke application.
      */
     public static void main(String[] args) {
-        String logo = "                  _                            _                                               \n" +
-                "                  (_)                          | |                                              \n" +
-                "   ___  __ _ _   _ _ _ __  _ __ ___   ___ _ __ | |_ _ __ ___   __ _ _ __   __ _  __ _  ___ _ __ \n" +
-                "  / _ \\/ _` | | | | | '_ \\| '_ ` _ \\ / _ \\ '_ \\| __| '_ ` _ \\ / _` | '_ \\ / _` |/ _` |/ _ \\ '__|\n" +
-                " |  __/ (_| | |_| | | |_) | | | | | |  __/ | | | |_| | | | | | (_| | | | | (_| | (_| |  __/ |   \n" +
-                "  \\___|\\__, |\\__,_|_| .__/|_| |_| |_|\\___|_| |_|\\__|_| |_| |_|\\__,_|_| |_|\\__,_|\\__, |\\___|_|   \n" +
-                "          | |       | |                                                          __/ |          \n" +
-                "          |_|       |_|                                                         |___/";
-        System.out.println("Hello from\n" + logo);
-        System.out.println("What do you want to do?");
-
-        new Duke().runCommandLoop();
+        Duke duke = new Duke();
+
+        duke.start();
+        duke.runCommandLoop();
+    }
+
+    /**
+     * Initialises the required objects and loads up the data from storage and show welcome message to user
+     */
+    private void start() {
+        ui = new TextUi();
+        ui.showWelcomeMessage();
     }
 
     private void runCommandLoop() {
         Scanner in = new Scanner(System.in);
-        String userCommand = in.nextLine();
+        String userCommand;
         Command command;
         CommandResult result;
-        while (!userCommand.equals("bye")) {
+        do {
+            userCommand = in.nextLine();
             command = new Parser().parseCommand(userCommand);
```

Comment:
Would it be better for the Parser object to be outside of the while loop? If it's here, will we be creating many iterations of it?